e2e-tests/test/transactions.js

```diff
@@ -1,70 +1,94 @@
-        assert(res.error && !res.result, "Valid transaction");
-        assert(res.error.message === "Extrinsic has invalid format: Error decoding field OpaqueExtrinsic.0");
+    it("should throw on missing to", async() =>{
+        const nonce = await node1.getNonce(node1Obj.publicKey);
+        const nextNonce = new U64(TypeRegistry, nonce.toNumber() + 1);
+        const res = await node1.unsafeTransfer("0x", 10021, nextNonce, node1Obj.privateKey);
+        assert(res.error && !res.result, "Valid transaction: Transaction did not throw");
+        assert(res.error.message === "Verification Error: Execution: Trap: Trap { kind: Unreachable }");
+    })
+    it("should throw on missing amount", async() =>{
+        const nonce = await node1.getNonce(node1Obj.publicKey);
+        const nextNonce = new U64(TypeRegistry, nonce.toNumber() + 1);
+        const res = await node1.unsafeTransfer(node3Obj.publicKey, null, nextNonce, node1Obj.privateKey);
+        assert(res.error && !res.result, "Valid transaction: Transaction did not throw");
+        assert(res.error.message === "Verification Error: Execution: Trap: Trap { kind: Unreachable }");
+    })
+    it("should throw on missing nonce", async() =>{
+        const res = await node1.unsafeTransfer(node2Obj.publicKey, 10021, null, node1Obj.privateKey);
+        assert(res.error && !res.result, "Valid transaction: Transaction did not throw");
+        assert(res.error.message === "Verification Error: Execution: Trap: Trap { kind: Unreachable }");
     })
 
     it("tx should be mined, i.e balances reduced/increased correctly", async ()=>{
-        const senderOldBalance = await node1.getBalance(nodes[0].publicKey);
-        const receiverOldBalance = await node1.getBalance(nodes[2].publicKey);
-        const res = await node1.transfer(nodes[2].publicKey, 1000, nodes[0].privateKey);
+        const senderOldBalance = await node1.getBalance(node1Obj.publicKey);
+        const receiverOldBalance = await node1.getBalance(node3Obj.publicKey);
+        const res = await node1.transfer(node3Obj.publicKey, 69, node1Obj.privateKey);
         assert(!res.error && res.result, "Invalid transaction");
         node1.waitForHead().then(async() => {
-            const senderNewBalance = await node1.getBalance(nodes[0].publicKey);
-            const receiverNewBalance = await node1.getBalance(nodes[2].publicKey);
+            const senderNewBalance = await node1.getBalance(node1Obj.publicKey);
+            const receiverNewBalance = await node1.getBalance(node3Obj.publicKey);
+            const amount = new U128(TypeRegistry, 69);
             assert(senderOldBalance != senderNewBalance && receiverOldBalance != receiverNewBalance, 
                 "Transaction not applied"
             );
+
+            assert(senderOldBalance.sub(amount).eq(senderNewBalance) && receiverOldBalance.add(amount).eq(receiverNewBalance), 
+                "Transaction not applied");
```

Comment:
- Imo it would be better if we have 2 separate `asserts`. If something fails, we would like to know what exactly failed and not guess whether the sender balance was not decreased or the receiver balance was not increased.

- The message I think should be more like: "Sender balance was not decreased correctly" or "Receiver balance was not increased correctly"